Source/Core/Prune/Util.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Boogie
+{
+  public static class Util
+  {
+    public static V GetOrCreate<K, V>(this IDictionary<K, V> dictionary, K key, Func<V> createValue)
+    {
+      if (dictionary.TryGetValue(key, out var result)) {
+        return result;
+      }
+
+      result = createValue();
+      dictionary[key] = result;
+      return result;
+    }
+    
+    public static uint SafeMult(uint a, uint b) {
+      uint result = UInt32.MaxValue;
+      try {
+        checked {
+          result = a * b;
+        }
+      } catch (OverflowException) { }
+
+      return result;
```

Comment:
I think this would be clearer as

```
try {
  return a * b;
} catch (OverflowException) {
  return UInt32.MaxValue;
}
```

Reply:
This was existing code copied from somewhere else, but I'll change it.